pkg/process/cluster_tracking.go

```diff
@@ -0,0 +1,115 @@
+package process
+
+import (
+	"sort"
+	"time"
+
+	kebruntime "github.com/kyma-project/kyma-environment-broker/common/runtime"
+)
+
+type runtimeState int
+
+const (
+	provisioning runtimeState = iota
+	deprovisioning
+	upgradingkyma
+	upgradingcluster
+	update
+	suspension
+	unsuspension
+)
+
+type simpleOperation struct {
```

Comment:
Are there non-simple operations planned?

Reply:
nope, there are only the complex ones that we inherited from key